File: src/Organization/RegOS.Organization.Domain/Aggregates/OrganizationSite/OrganizationSiteType.cs
namespace RegOS.Organization.Domain.Aggregates.OrganizationSite;

/// <summary>
/// What a site is for.
/// </summary>
/// <remarks>
/// A closed enum rather than reference data: it drives behaviour. Only a
/// manufacturing site can be named on a licence as an approved manufacturer,
/// and only a testing site can be a release-testing location — rules that live
/// in code, so the vocabulary they branch on does too. Contrast
/// <c>IdentifierScheme</c>, which nothing branches on and is therefore data.
/// </remarks>
public enum OrganizationSiteType
{
    Manufacturing = 1,
    Packaging = 2,
    Testing = 3,
    Storage = 4,

    /// <summary>An office of a health authority — where correspondence goes.</summary>
    AuthorityOffice = 5,

    /// <summary>A sponsor or partner's own office, not a production location.</summary>
    Office = 6,
}

File: web/regos-web/src/features/regulatory/organizations/types/OrganizationSiteType.ts
/**
 * Mirrors OrganizationSiteType on the server, which is a closed enum rather
 * than reference data because rules branch on it — only a Manufacturing site
 * can be named on a licence as an approved manufacturer.
 *
 * Kept as an explicit list rather than derived from returned rows: the create
 * form has to offer every type, including ones this tenant has never used.
 */
export const ORGANIZATION_SITE_TYPES = [
  { value: "Manufacturing", label: "Manufacturing" },
  { value: "Packaging", label: "Packaging" },
  { value: "Testing", label: "Testing" },
  { value: "Storage", label: "Storage" },
  { value: "AuthorityOffice", label: "Authority Office" },
  { value: "Office", label: "Office" },
] as const;

export function siteTypeLabel(value: string): string {
  return (
    ORGANIZATION_SITE_TYPES.find((type) => type.value === value)?.label ?? value
  );
}
